src/code/z_shrink_window.c

```diff
@@ -1,25 +1,121 @@
+/**
+ * File: z_shrink_window.c
+ * Description: Draws black top/bottom/side borders on the viewing window (e.g. Z-targeting)
+ */
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_SetLetterboxTarget.s")
+ShrinkWindowContext gShrinkWindowContext;
+ShrinkWindowContext* gShrinkWindowContextPtr;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetLetterboxTarget.s")
+void ShrinkWindow_SetLetterboxTarget(s32 target) {
+    gShrinkWindowContextPtr->letterboxTarget = target;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_SetLetterboxMagnitude.s")
+s32 ShrinkWindow_GetLetterboxTarget(void) {
+    return gShrinkWindowContextPtr->letterboxTarget;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetLetterboxMagnitude.s")
+void ShrinkWindow_SetLetterboxMagnitude(s32 magnitude) {
+    gShrinkWindowContextPtr->letterboxMagnitude = magnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_SetPillarboxTarget.s")
+s32 ShrinkWindow_GetLetterboxMagnitude(void) {
+    return gShrinkWindowContextPtr->letterboxMagnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetPillarboxTarget.s")
+void ShrinkWindow_SetPillarboxTarget(s32 target) {
+    gShrinkWindowContextPtr->pillarboxTarget = target;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_SetPillarboxMagnitude.s")
+s32 ShrinkWindow_GetPillarboxTarget(void) {
+    return gShrinkWindowContextPtr->pillarboxTarget;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetPillarboxMagnitude.s")
+void ShrinkWindow_SetPillarboxMagnitude(s32 magnitude) {
+    gShrinkWindowContextPtr->pillarboxMagnitude = magnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Init.s")
+s32 ShrinkWindow_GetPillarboxMagnitude(void) {
+    return gShrinkWindowContextPtr->pillarboxMagnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Fini.s")
+void ShrinkWindow_Init(void) {
+    gShrinkWindowContextPtr = &gShrinkWindowContext;
+    bzero(gShrinkWindowContextPtr, sizeof(gShrinkWindowContext));
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Step.s")
+void ShrinkWindow_Destroy(void) {
+    gShrinkWindowContextPtr = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Draw.s")
+void ShrinkWindow_Update(s32 framerateDivisor) {
+    s32 step;
+    s32 nextMagnitude;
+
+    step = ((framerateDivisor == 3) ? 10 : 30 / framerateDivisor );
+
+    nextMagnitude = gShrinkWindowContextPtr->letterboxMagnitude;
```

Comment:
```suggestion
void ShrinkWindow_Update(s32 framerateDivisor) {
    s32 step = ((framerateDivisor == 3) ? 10 : 30 / framerateDivisor );
    s32 nextMagnitude;

    nextMagnitude = gShrinkWindowContextPtr->letterboxMagnitude;
```
I'm not convinced this is better, I'll leave it up to you